File: src/main/java/gay/skitbet/jackiro/task/LoadGuildsTask.java
package gay.skitbet.jackiro.task;

import gay.skitbet.jackiro.Jackiro;
import gay.skitbet.jackiro.managers.MongoManager;
import gay.skitbet.jackiro.model.ServerConfig;
import gay.skitbet.jackiro.utils.JackiroEmbed;
import net.dv8tion.jda.api.entities.channel.concrete.TextChannel;

public class LoadGuildsTask extends Thread {

    private final Jackiro jackiro;

    public LoadGuildsTask(Jackiro jackiro) {
        super("Update Forgotten Guilds Task");
        this.jackiro = jackiro;
    }

    @Override
    public void run() {
        for (var guild : jackiro.getShardManager().getGuilds()) {
            ServerConfig config = MongoManager.getServerConfigRepository().findById(guild.getId());
            if (config == null) {
                config = new ServerConfig(guild.getId());
                MongoManager.getServerConfigRepository().save(config);
                Jackiro.LOGGER.info("Created default config for guild: " + guild.getName());

                if (guild.getDefaultChannel() instanceof TextChannel textChannel) {
                    textChannel.sendMessageEmbeds(JackiroEmbed.getNewGuildEmbed(guild)).queue();
                }
            } else {
                System.out.println("Loaded existed config for guild: " + guild.getName());
            }
        }
    }
}

File: src/main/kotlin/gay/skitbet/jackiro/task/LoadGuildsTask.kt
package gay.skitbet.jackiro.task

import gay.skitbet.jackiro.Jackiro
import gay.skitbet.jackiro.managers.MongoManager
import gay.skitbet.jackiro.model.ServerConfig
import gay.skitbet.jackiro.utils.JackiroEmbed
import net.dv8tion.jda.api.entities.channel.concrete.TextChannel
import java.util.concurrent.ScheduledExecutorService
import java.util.concurrent.TimeUnit

class LoadGuildsTask(private val jackiro: Jackiro) {

    fun start(scheduler: ScheduledExecutorService) {
        scheduler.schedule(::loadGuilds, 0, TimeUnit.SECONDS)
    }

    fun loadGuilds() {
        val shardManager = jackiro.shardManager

        for (guild in shardManager.guilds) {
            val repo = MongoManager.serverConfigRepository
            var config = repo.findById(guild.id)

            if (config == null) {
                config = ServerConfig(guild.id)
                repo.save(config)
                Jackiro.logger.info("Created default config for guild: ${guild.name}")

                (guild.defaultChannel as? TextChannel)?.let { channel ->
                    channel.sendMessageEmbeds(JackiroEmbed.getNewGuildEmbed(guild)).queue()
                }
            } else {
                Jackiro.logger.info("Loaded existing config for guild: ${guild.name}")
            }
        }
    }
}
